pallets/relayer/src/lib.rs

```diff
@@ -374,10 +363,23 @@ impl<T: Config> Module<T> {
             Error::<T>::PayingKeyCddMissing
         );
 
+        // Remove existing subsidy for the user_key, if it exists.
+        if let Some(subsidy) = <Subsidies<T>>::take(&user_key) {
```

Comment:
```suggestion
        if let Some(subsidy) = <Subsidies<T>>::get(&user_key) {
```
The later `::insert` should be sufficient.